ThunderBasics/TSCToastView.m

```diff
@@ -169,7 +173,14 @@ - (void)showWithCompletion:(TSCToastDisplayCompletion)completion
 
 - (void)layout
 {
-    CGRect labelContainer = CGRectMake(kToastEdgeInsets.left, kToastEdgeInsets.top, self.frame.size.width - kToastEdgeInsets.left - kToastEdgeInsets.right, MAXFLOAT);
+    CGRect labelContainer;
+    // Get safe insets so the toast isn't covered up by the notch on iPhone X (We can't use self.window because it's safeAreaInsets are zero)
+    if (@available(iOS 11.0, *)) {
+        UIEdgeInsets safeInsets = [UIApplication sharedApplication].keyWindow.rootViewController.view ? [UIApplication sharedApplication].keyWindow.rootViewController.view.safeAreaInsets : UIEdgeInsetsZero;
+        labelContainer = CGRectMake(kToastEdgeInsets.left, kToastEdgeInsets.top + safeInsets.top, self.frame.size.width - kToastEdgeInsets.left - kToastEdgeInsets.right, MAXFLOAT);
+    } else {
+        labelContainer = CGRectMake(kToastEdgeInsets.left, kToastEdgeInsets.top, self.frame.size.width - kToastEdgeInsets.left - kToastEdgeInsets.right, MAXFLOAT);
```

Comment:
I suppose instead of setting `labelContainer` in two places, you could have
```
UIEdgeInsets inset = UIEdgeInsetsZero;

if (@available(iOS 11.0, *)) {
    inset = [UIApplication sharedApplication].keyWindow.rootViewController.view ? [UIApplication sharedApplication].keyWindow.rootViewController.view.safeAreaInsets : UIEdgeInsetsZero;
}

CGRect labelContainer = CGRectMake(kToastEdgeInsets.left, kToastEdgeInsets.top + inset.top, self.frame.size.width - kToastEdgeInsets.left - kToastEdgeInsets.right, MAXFLOAT);
```

(or something along those lines :) )